File: modelo/menu.cs
using System.Data;
using System;
public class menu : oObjeto
{
    public string Nombre { get; set; }
    public string Componente { get; set; }
    public string Tipo { get; set; }
    public menu() { }
    public menu(DataRow dr)
    {
        this.Id = Convert.ToInt32(dr["id"]);
        this.Nombre = dr["nombre"].ToString();
        this.Componente = dr["componente"].ToString();
        this.Tipo = dr["tipo"].ToString();
         
    }

}

File: ClientApp/src/app/clases/menu.ts

export class menu {
    id : number;
    nombre : string;
    componente : string;
    tipo : string;

    constructor(obj:any){
        //el orden de os parametros afecta la vista de los input en el modal
        
        this.nombre = obj.nombre;
        this.componente = obj.componente
        this.tipo = obj.tipo;
        this.id = +obj.id; // formateo a number el string // parseInt(val
    }
}